### PR TITLE
Orchestration groups

### LINKED ISSUE #363: update Transaction Schema to include custom metrics and orchestration groups
@devcritter thoughts? about how to represent orchestration groups

### PR DESCRIPTION
**Pull request by debonair**
@rcrichton please review closes #363

### CODE REVIEW

File: src/middleware/stats.coffee [hunk 1]
```diff
@@ -41,26 +41,29 @@ exports.incrementTransactionCount = (ctx, done) ->
...
-                  logger.info 'incrementing '+ route.name + 'orchestration timer ' + metric.name
-                  sdc.timing      domain + '.channels.' + ctx.authorisedChannel._id + '.nonPrimaryRoutes.' + route.name + '.orchestrations.' + orchestration.name + '.' + metric.name, metric.value
-
-                if metric.type == 'gauge'
-                  logger.info 'incrementing '+ route.name + 'orchestration gauge ' + metric.name
-                  sdc.gauge       domain + '.channels.' + ctx.authorisedChannel._id + '.nonPrimaryRoutes.' + route.name + '.orchestrations.' + orchestration.name  + '.' + metric.name, metric.value
+            do (orchestration) ->
+              orchestrationStatus = orchestration.response.status
+              if orchestration.group
+                orchestrationName = "#{orchestration.group}.#{orchestration.name}" #Namespace it by group
```
Comment (rcrichton) [inline src/middleware/stats.coffee]: If there is no group how does orchestration name get set?

File: test/unit/statsMiddlewareTest.coffee [hunk 2]
```diff
@@ -139,14 +141,14 @@ describe "Stats Middleware ", ->
...
-            s.expectMessage domain + '.channels.ckjhfjwedsnfdsf.orchestrations.Lab API.statusCodes.200:5|ms', ->
+          s.expectMessage domain + '.channels.ckjhfjwedsnfdsf.orchestrations.group.Lab API:5|ms', ->
+            s.expectMessage domain + '.channels.ckjhfjwedsnfdsf.orchestrations.group.Lab API.statusCodes.200:5|ms', ->
               s.expectMessage domain + '.channels.Successful:10|ms', ->
                 s.expectMessage domain + '.channels.ckjhfjwedsnfdsf:10|ms', ->
                   s.expectMessage domain + '.channels.ckjhfjwedsnfdsf.statuses.Successful:10|ms', ->
                     s.expectMessage domain + '.channels.ckjhfjwedsnfdsf.nonPrimaryRoutes.secondary route:10|ms', ->
                       s.expectMessage domain + '.channels.ckjhfjwedsnfdsf.nonPrimaryRoutes.secondary route.statusCodes.200:10|ms', ->
-                        s.expectMessage domain + '.channels.ckjhfjwedsnfdsf.nonPrimaryRoutes.secondary route.orchestrations.Lab API:5|ms', done
+                        s.expectMessage domain + '.channels.ckjhfjwedsnfdsf.nonPrimaryRoutes.secondary route.orchestrations.group.Lab API:5|ms', done
```
Comment (rcrichton) [inline test/unit/statsMiddlewareTest.coffee]: I think we need test for when group is set and when it isn't set.

**Comment by rcrichton** [comment]
Hey @debonair this looks good. Just a few comments above.

File: test/unit/statsMiddlewareTest.coffee (hunk 2 above)
Comment (debonair) [inline test/unit/statsMiddlewareTest.coffee]: OK I overlooked that lemme add in the extra tests

**Comment by debonair** [comment]
@rcrichton could you eyeball these changes before I merge the branch in

**Comment by rcrichton** [comment]
Thanks @debonair this looks good to me :+1:

**Pull request merged**